src/RiakClient/RiakClient.cs

```diff
@@ -569,16 +571,54 @@ public RiakResult<RiakStreamedMapReduceResult> StreamMapReduce(RiakMapReduceQuer
         /// physical I/O and can take a long time.</remarks>
         public RiakResult<IEnumerable<string>> ListBuckets()
         {
-            var result = UseConnection(conn => conn.PbcWriteRead<RpbListBucketsResp>(MessageCode.RpbListBucketsReq));
+            return ListBuckets(RiakConstants.DefaultBucketType);
+        }
+
+        /// <summary>
+        /// Lists all buckets in the specified Bucket Type.
+        /// </summary>
+        /// <param name="bucketType">The name of the Bucket Type to list buckets for.</param>
+        /// <returns>
+        /// An <see cref="System.Collections.Generic.IEnumerable&lt;T&gt;"/> of <see cref="string"/> bucket names.
+        /// </returns>
+        /// <remarks>Buckets provide a logical namespace for keys. Listing buckets requires folding over all keys in a cluster and 
+        /// reading a list of buckets from disk. This operation, while non-blocking in Riak 1.0 and newer, still produces considerable
+        /// physical I/O and can take a long time.</remarks>
+        public RiakResult<IEnumerable<string>> ListBuckets(string bucketType) 
+        {            
+            WarnAboutListBuckets();
+
+            var listBucketReq = new RpbListBucketsReq
+            {
+                type = bucketType.ToRiakString()
+            };
+
+            var result = UseConnection(conn => conn.PbcWriteRead<RpbListBucketsReq, RpbListBucketsResp>(listBucketReq));
 
             if (result.IsSuccess)
             {
-                var buckets = result.Value.buckets.Select(b => b.FromRiakString());
-                return RiakResult<IEnumerable<string>>.Success(buckets.ToList());
+                var buckets = result.Value.buckets.Select(b => b.FromRiakString()).ToList();
```

Comment:
Is `.ToList()` necessary? I wonder what the "best practice" is for API clients that have enumerable results.